internal/events/persist_batch.go

```diff
@@ -143,27 +133,45 @@ func (em *eventManager) validateAndPersistBatchContent(ctx context.Context, batc
 	return em.persistBatchContent(ctx, batch, matchedMsgs)
 }
 
-func (em *eventManager) validateBatchData(ctx context.Context, batch *fftypes.Batch, i int, data *fftypes.Data) bool {
+func (em *eventManager) validateBatchData(ctx context.Context, batch *fftypes.Batch, i int, data *fftypes.Data) (bool, error) {
 
 	l := log.L(ctx)
 	l.Tracef("Batch '%s' data %d: %+v", batch.ID, i, data)
 
 	if data == nil {
 		l.Errorf("null data entry %d in batch '%s'", i, batch.ID)
-		return false
+		return false, nil
 	}
 
 	hash, err := data.CalcHash(ctx)
 	if err != nil {
 		log.L(ctx).Errorf("Invalid data entry %d in batch '%s': %s", i, batch.ID, err)
-		return false
+		return false, nil
 	}
 	if data.Hash == nil || *data.Hash != *hash {
 		log.L(ctx).Errorf("Invalid data entry %d in batch '%s': Hash=%v Expected=%v", i, batch.ID, data.Hash, hash)
-		return false
+		return false, nil
 	}
 
-	return true
+	if data.Blob != nil && batch.Type == fftypes.BatchTypeBroadcast {
```

Comment:
Maybe just semantics, but it felt surprising for the download to be triggered from a `validate` method

Reply:
👍  - have split the function in two